src/App.tsx

```diff
@@ -7,35 +7,54 @@ import LoginPage from "./Components/login/LoginPage";
 import LoginEmailPass from "./Components/update/LoginEmailPass";
 import UpdatePassword from "./Components/update/UpdatePassword";
 import Layout from "./layout/Layout";
-import { LOCAL_STORAGE_KEYS, PAGES } from "./utils/constants/constants";
+import { PAGES } from "./utils/constants/constants";
 import Basket from "./Components/basket/Basket";
+import CheckOut from "./Components/CheckOut/CheckOut";
+import { IBasketProduct } from "./utils/interface";
+import { MyGlobalContext } from "./Components/home/SingleProduct/components/context/useClobalContext";
 
 function App() {
-  const navigate = useNavigate();
-
-  useEffect(() => {
-    (async () => {
-      if (!localStorage.getItem(LOCAL_STORAGE_KEYS.JWT_TOKEN)) {
-        navigate(PAGES.LOGIN);
-      }
-    })();
-  }, [navigate]);
 
+  const [basket, setBasket] = useState<IBasketProduct[]>([]);
```

Comment:
```suggestion
  const [basket, setBasket] = useState<IBasketProduct[]>([]);
```

switch this to redux + redux actions(redux think or redux saga)

Reply:
after that, move all App.tsx unrelated logic from this file.